### PR TITLE
Some test plan for integration feature

### PR DESCRIPTION
**Pull request by SagarGi**
## Description
This PR adds some of the somke testing to be done manually during the time of the major, minor and patch (as per requirement) release of the integration application. The need for this test plan is because our application lacks end to end integration testing where both `Nextcloud` and `OpenProject` instances are involved.

## Related Issue or Workpackage

- Fixes https://community.openproject.org/projects/nextcloud-integration/work_packages/52631/activity

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Updated `CHANGELOG.md` file

### CODE REVIEW

File: docs/testing/smoke_testing.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+### 6. Link/Unlink a work package for a file/folder in Nextcloud
+- [ ] Complete Smoke Test No 1.
+- [ ] Complete Smoke Test No 2.
+- [ ] Complete Smoke Test No 3.
+- [ ] Select a file, navigate to sidebar `OpenProject` tab.
+- [ ] Search for any of the work packages in the `Demo Project`.
+- [ ] Work packages are listed.
+- [ ] Link to any one of the work packages appeared in the search lists.
+- [ ] Linked work package is appeared in the `OpenProject` Tab with a toast message.
```
Comment (phil-davis) [inline docs/testing/smoke_testing.md:54]:
```suggestion
- [ ] Linked work package appears in the `OpenProject` Tab with a toast message.
```

I don't know what a "toast message" is, but I suppose I would need to do the test steps and then I will see it.

File: docs/testing/smoke_testing.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+- [ ] Link other work packages, refresh the `Nextcloud` instance and all the linked ones should appear in the `OpenProject` Tab.
```
Comment (phil-davis) [inline docs/testing/smoke_testing.md:55]: "refresh the Nextcloud instance - I suppose you mean "reload the browser"?

File: docs/testing/smoke_testing.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+- [ ] Hover to a work package to be unlinked, unlink button is visible.
+- [ ] Unlink a work package and should be deleted from the `OpenProject` Tab with a toast message.
```
Comment (phil-davis) [inline docs/testing/smoke_testing.md:57]:
```suggestion
- [ ] Unlink a work package and it should be deleted from the `OpenProject` Tab with a toast message.
```

File: docs/testing/smoke_testing.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+### 7. Link/Unlink a work package for a file/folder from OpenProject
+- [ ] Complete Smoke Test No 1.
+- [ ] Complete Smoke Test No 3.
+- [ ] Complete Smoke Test No 4.
+- [ ] Navigate to `Demo Project > Work Packages` and double click any one of the work packages available.
+- [ ] Navigate to `Files` tab, click `link existing files`, select available files (for e.g welcome.txt) from nextcloud and link it to the work package.
```
Comment (phil-davis) [inline docs/testing/smoke_testing.md:64]:
```suggestion
- [ ] Navigate to `Files` tab, click `link existing files`, select available files (for e.g welcome.txt) from nextcloud and link it to the work package.
```
```suggestion
- [ ] Navigate to `Files` tab, click `link existing files`, select available files (for example, welcome.txt) from Nextcloud and link it to the work package.
```

File: docs/testing/smoke_testing.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+- [ ] Selected file is linked to the work package in `OpenProject`
+- [ ] Also Navigate to nextcloud and see in the `OpenProject` tab for file (welcome.txt), the work package should be linked.
+
+### 8. Direct upload file/folder from OpenProject to Nextcloud
+- [ ] Complete Smoke Test No 1.
+- [ ] Complete Smoke Test No 3.
+- [ ] Complete Smoke Test No 4.
+- [ ] Navigate to `Demo Project > Work Packages` and double click any one of the work packages available.
+- [ ] Navigate to `Files` tab, click `Upload files`, select available files from your local system (for e.g local.txt) and upload choosing the upload location.
```
Comment (phil-davis) [inline docs/testing/smoke_testing.md:73]:
```suggestion
- [ ] Navigate to `Files` tab, click `Upload files`, select available files from your local system (for e.g local.txt) and upload choosing the upload location.
```
```suggestion
- [ ] Navigate to `Files` tab, click `Upload files`, select available files from your local system (for example, local.txt) and upload choosing the upload location.
```

Either write "for example" or "e.g."

File: docs/testing/smoke_testing.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+- [ ] Uploaded file is linked to the work package in `OpenProject`
+- [ ] Also Navigate to `Nextcloud` and see in the `OpenProject` tab for file (local.txt), the work package should be linked.
+
+### 9. Create a WorkPackage from the Nextcloud
+- [ ] Complete Smoke Test No 1.
+- [ ] Complete Smoke Test No 2.
+- [ ] Complete Smoke Test No 3.
+- [ ] Open the form to create work package from the nextcloud in `OpenProject` tab for a file/folder.
```
Comment (phil-davis) [inline docs/testing/smoke_testing.md:81]:
```suggestion
- [ ] Open the form to create work package from Nextcloud in the `OpenProject` tab for a file/folder.
```

File: docs/testing/smoke_testing.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+### 10. Check OpenProject Notification
+> Make sure your `OpenProject` is running along with `worker` instance
+- [ ] Complete Smoke Test No 1.
+- [ ] Complete Smoke Test No 2.
+- [ ] Complete Smoke Test No 3.
+- [ ] Complete Smoke Test No 4.
+- [ ] Create a separate user in both `Nextcloud` as well as `OpenProject`.
+- [ ] Connect `Nextcloud` user to `OpenProject` user and vice-versa (`OpenProject` user to `Nextcloud` user).
+- [ ] Now as an `OpenProject` admin, assign any of the `Demo Project` work package to created `OpenProject` user.
```
Comment (phil-davis) [inline docs/testing/smoke_testing.md:93]:
```suggestion
- [ ] Now as an `OpenProject` admin, assign any of the `Demo Project` work package to created `OpenProject` user.
```
```suggestion
- [ ] Now as an `OpenProject` admin, assign any of the `Demo Project` work packages to the created `OpenProject` user.
```

File: docs/testing/smoke_testing.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+### 11. Check New folder with automatically managed permissions in OpenProject
+- [ ] Complete Smoke Test No 1.
+- [ ] Complete Smoke Test No 3 (Make sure to choose `New folder with automatically managed permissions` while creating `File storage`).
+- [ ] Complete Smoke Test No 4.
+- [ ] Navigate to `Demo Project > Work Packages` and double click any one of the work packages available.
+- [ ] Navigate to `Files` tab, click `link existing files`.
+- [ ] In a modal, `Nextcloud > OpenProject > Demo project(1)` should be visible.
+- [ ] Also Navigate to `Nextcloud` and in Files `OpenProject > Demo project(1)` folder is created.
+- [ ] Try to delete `OpenProject` or `OpenProject > Demo project(1)`, shoudl not be deleted.
```
Comment (phil-davis) [inline docs/testing/smoke_testing.md:104]:
```suggestion
- [ ] Try to delete `OpenProject` or `OpenProject > Demo project(1)`. They should not be deleted. 
```

File: docs/testing/smoke_testing.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+- [ ] Navigate to `Files` tab, click `link existing files`.
+- [ ] In a modal, `Nextcloud > OpenProject > Demo project(1)` should be visible.
+- [ ] Also Navigate to `Nextcloud` and in Files `OpenProject > Demo project(1)` folder is created.
+- [ ] Try to delete `OpenProject` or `OpenProject > Demo project(1)`, shoudl not be deleted. 
+
+### 12. Check the integration script
+
+> Before Running the script make sure that your `Nextcloud` and `OpenProject` instance is up and running
+
+- [ ] Run the `integration_setup.sh` script with to set up integration without project folder with following command:
```
Comment (phil-davis) [inline docs/testing/smoke_testing.md:110]:
```suggestion
- [ ] Run the `integration_setup.sh` script with to set up integration without project folder with the following command:
```

File: docs/testing/smoke_testing.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+NEXTCLOUD_HOST=<nextcloud_instance_url> \
+OPENPROJECT_HOST=<openproject_instance_url> \
+OP_ADMIN_USERNAME=admin \
+OP_ADMIN_PASSWORD=admin \                                                                                                             
+NC_ADMIN_USERNAME=admin \                                                                                     
+NC_ADMIN_PASSWORD=admin \                                                             
+OPENPROJECT_STORAGE_NAME=Nextcloud  \                         
+bash integration_setup.sh
+```
+- [ ] Upon successful, try Smoke Test No 2 (Skip first check).
```
Comment (phil-davis) [inline docs/testing/smoke_testing.md:122]:
```suggestion
- [ ] Upon success, try Smoke Test No 2 (Skip first check).
```

File: docs/testing/smoke_testing.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+- [ ] Upon successful, try Smoke Test No 4 (Skip first check).
```
Comment (phil-davis) [inline docs/testing/smoke_testing.md:123]:
```suggestion
- [ ] Upon success, try Smoke Test No 4 (Skip first check).
```

File: docs/testing/smoke_testing.md (hunk 2 above)
Comment (SagarGi) [reply docs/testing/smoke_testing.md:55]: Yes exactly. Thanks!

File: docs/testing/smoke_testing.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+- [ ] Navigate to `Administration > File storages` in `OpenProject`.
+- [ ] Create a file storage type `Nextcloud` and name it as `Nextcloud` in `OpenProject`.
+- [ ] Navigate to `Administration Settings > OpenProject` in admin setting of `Nextcloud`.
+- [ ] Copy `OpenProject` Oauth Credential (client_id and client_secret) and save them in `Nextcloud`.
+- [ ] Copy `Nextcloud` Oauth Credential (client_id and client_secret) and save them in `OpenProject`.
+
+### 2. Connect Nextcloud with OpenProject (Without project folder setup)
+- [ ] Complete Smoke Test No 1.
+- [ ] In `Administration Settings > OpenProject` personal setting of `Nextcloud` admin, connect to `OpenProject`.
+- [ ] `Nextcloud` amdin should be connected as an `OpenProject` admin.
```
Comment (nabim777) [inline docs/testing/smoke_testing.md:15]:
```suggestion
- [ ] `Nextcloud` admin should be connected as an `OpenProject` admin.
```

File: docs/testing/smoke_testing.md [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+- [ ] Also, create a user in both `Nextcloud` as well as `OpenProject`.
+- [ ] From the personal section of the created user in `Nextcloud`, connect to `OpenProject`.
+- [ ] `Nextcloud` user should be connected as an `OpenProject` user.
+
+### 3. Add File storage (Nextcloud) to an OpenProject project
+- [ ] Complete Smoke Test No 1.
+- [ ] Select an `OpenProject` Project (e.g. `Demo Project`) in `OpenProject`.
+- [ ] Navigate to `Project setings > modules` of `Demo Project`.
+- [ ] In modules, check `File storages`, choose `No specific Folder` option and save the setting.
+- [ ] Now `Project setings > File storages` option is visible
```
Comment (nabim777) [inline docs/testing/smoke_testing.md:25]:
```suggestion
- [ ] Now `Project settings > File storages` option is visible
```

File: docs/testing/smoke_testing.md [hunk 14]
```diff
@@ -0,0 +1,125 @@
+This file consists of some smoke testing to be done before integration application release (major and minor).
```
Comment (nabim777) [inline docs/testing/smoke_testing.md:1]:
```suggestion
This file consists of some smoke testing to be done before integration_openproject application release (major and minor).
```

File: docs/testing/smoke_testing.md [hunk 15]
```diff
@@ -0,0 +1,125 @@
+This file consists of some smoke testing to be done before the release (major and minor) of `integration_application` application.
+The need for this smoke testing (manual) is that we do not have e2e test setup to automate that involves both `OpenProject` and `Nextcloud`.
+
+## Smoke Test for `integration_openproject` (In Nextcloud and OpenProject)
+### 1. Oauth configuration (without project folder setup/Automatically managed folders)
+- [ ] Navigate to `Administration > File storages` in `OpenProject`.
```
Comment (individual-it) [inline docs/testing/smoke_testing.md:6]: What about turning the sentences around, saying first where the action should be done?
```suggestion
- [ ] In `OpenProject` navigate to `Administration > File storages`.
```

File: docs/testing/smoke_testing.md [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+
+### 9. Create a WorkPackage from the Nextcloud
+- [ ] Complete Smoke Test No 1.
+- [ ] Complete Smoke Test No 2.
+- [ ] Complete Smoke Test No 3.
+- [ ] Open the form to create work package from Nextcloud in the `OpenProject` tab for a file/folder.
+- [ ] Select `Demo Project`, fill up the modal form and create.
+- [ ] Work package should be created and linked to the selected file.
+
+### 10. Check OpenProject Notification
```
Comment (individual-it) [inline docs/testing/smoke_testing.md:85]: Does this functionality rely on the integration app, or is it purely an OpenProject functionality?

**Review by individual-it — CHANGES_REQUESTED** [review]

**Comment by individual-it** [comment]
also the release documentation should be adjusted: https://github.com/nextcloud/integration_openproject/blob/master/docs/release.md

File: docs/testing/smoke_testing.md (hunk 16 above)
Comment (SagarGi) [reply docs/testing/smoke_testing.md]: Its the notification that we get in the `openproject` widget in nextcloud for a user

**Review by individual-it — APPROVED** [review]

File: docs/testing/smoke_testing.md [hunk 17]
```diff
@@ -0,0 +1,125 @@
...
+- [ ] Navigate to `Files` tab, click `link existing files`.
+- [ ] In a modal, `Nextcloud > OpenProject > Demo project(1)` should be visible.
+- [ ] Also Navigate to `Nextcloud` and in Files `OpenProject > Demo project(1)` folder is created.
+- [ ] Try to delete `OpenProject` or `OpenProject > Demo project(1)`. They should not be deleted.
+
+### 12. Check the integration script
+
+> Before Running the script make sure that your `Nextcloud` and `OpenProject` instance is up and running
+
+- [ ] Run the `integration_setup.sh` script with to set up integration without project folder with the following command:
```
Comment (phil-davis) [inline docs/testing/smoke_testing.md:110]:
```suggestion
- [ ] Run the `integration_setup.sh` script to set up integration without project folder with the following command:
```

File: docs/testing/smoke_testing.md [hunk 18]
```diff
@@ -0,0 +1,125 @@
...
+### 8. Direct upload file/folder from OpenProject to Nextcloud
+- [ ] Complete Smoke Test No 1.
+- [ ] Complete Smoke Test No 3.
+- [ ] Complete Smoke Test No 4.
+- [ ] Navigate to `Demo Project > Work Packages` and double click any one of the work packages available.
+- [ ] Navigate to `Files` tab, click `Upload files`, select available files from your local system (for example, local.txt) and upload choosing the upload location.
+- [ ] Uploaded file is linked to the work package in `OpenProject`
+- [ ] Also Navigate to `Nextcloud` and see in the `OpenProject` tab for file (local.txt), the work package should be linked.
+
+### 9. Create a WorkPackage from the Nextcloud
```
Comment (phil-davis) [inline docs/testing/smoke_testing.md:77]:
```suggestion
### 9. Create a WorkPackage from Nextcloud
```

**Review by phil-davis — APPROVED** [review]

**Pull request merged**